src/Query/EloquentQueryBuilder.php

```diff
@@ -314,7 +315,7 @@ public function orWhereTime($column, $operator, $value = null)
 
     public function whereNested(Closure $callback, $boolean = 'and')
     {
-        $query = (new (get_class($this))(clone $this->builder));
```

Comment:
If this doesn’t pass builder then it won’t be defined. Does calling it this way call the constructor with the original params?

Reply:
Yep I understand it will make the class, but I dont think it will pass the params (in this case $builder) which is required. Surely it needs to be something like:

`app(static::class)(clone $this->builder)`

Sorry, mobile this week so dont have a huge amount of time to actually debug and test things otherwise I'd resolve myself.

Reply:
That's the beauty of Laravel's service container handling - the builder is automatically created and passes to the constructor

Reply:
We should actually be using `Illuminate\Contracts\Database\Eloquent\Builder` for the binding, instead of the actual builder class when I think about it - I'll just test if that works too.

Reply:
@ryanmitchell you can test it by changing this in your `composer.json` for the eloquent driver package:
```json
"repositories": [
    {
        "type": "vcs",
        "url": "https://github.com/frittenkeez/cms.git"
    }
],
"require": {
    "php": "^8.0",
    "statamic/cms": "dev-feature/eloquent-ors"
},
```